File: artifacts/littlefs/decompose/src/lfs_dir_t.c
typedef struct lfs_dir
{
  struct lfs_dir *next;
  uint16_t id;
  uint8_t type;
  lfs_mdir_t m;
  lfs_off_t pos;
  lfs_block_t head[2];
} lfs_dir_t
/*** DEPENDENCIES:
typedef uint32_t lfs_block_t
----------------------------
typedef struct lfs_mdir
{
  lfs_block_t pair[2];
  uint32_t rev;
  lfs_off_t off;
  uint32_t etag;
  uint16_t count;
  bool erased;
  bool split;
  lfs_block_t tail[2];
} lfs_mdir_t
----------------------------
typedef uint32_t lfs_off_t
----------------------------
***/

File: artifacts/littlefs/translate/src/lfs_dir_t.rs
use crate::*;
use std::clone::Clone;


pub type lfs_block_t = u32;
pub type lfs_off_t = u32;

#[derive(Clone)]
pub struct LfsDir {
    pub next: Option<Box<LfsDir>>,
    pub id: u16,
    pub type_: u8,
    pub m: LfsMdir,
    pub pos: lfs_off_t,
    pub head: [lfs_block_t; 2],
}

#[derive(Clone)]
pub struct LfsMdir {
    pub pair: [lfs_block_t; 2],
    pub rev: u32,
    pub off: lfs_off_t,
    pub etag: u32,
    pub count: u16,
    pub erased: bool,
    pub split: bool,
    pub tail: [lfs_block_t; 2],
}

